gh.go

```diff
@@ -107,6 +108,35 @@ func GQLClient(opts *api.ClientOptions) (api.GQLClient, error) {
 	return iapi.NewGQLClient(opts.Host, opts), nil
 }
 
+// HTTPClient builds a client that can be passed to another library.
+// As part of the configuration a hostname, auth token, and default set of headers are resolved
+// from the gh environment configuration. These behaviors can be overridden using the opts argument.
```

Comment:
We should add some comments about how providing the host option here will not actually direct a request to that host which is different from the other clients. Also, we should add some comments about how the auth token is protected from being sent to arbitrary hosts.

Reply:
Sounds sensible to me. Commit incoming.